examples/dom_example.re

```diff
@@ -2,16 +2,91 @@ open ReasonJs;
 
 /* Adapted from https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Examples#Example_7:_Displaying_Event_Object_Properties */
 
-open Document;
-open Element;
+
+/* Option.map */
+let map f => fun
+| Some v => Some (f v)
+| None => None;
+
+let and_then f => fun
+| Some v => f v
+| None => None;
+
+let unwrapUnsafely = fun
+| Some v => v
+| None => assert false;
```

Comment:
I've been thinking of having functions like `HtmlDivElement.create` to avoid the `option`, but maybe a GADT would be better.... Could perhaps be used with getElementsByTagName and such too...